utility/middleware.js

```diff
@@ -0,0 +1,61 @@
+import { externalApiCall } from "./externalApiCall.js"
+import { sendResponse } from "../utility/response.js";
+import { getUserByEmail } from "../model/users.js";
+import constant from "../constants/constant.js";
+
+const uri = "https://www.googleapis.com/oauth2/v1/userinfo"
+
+export async function verifyGoogleToken(req, res, next) {
+    const token = getTokenFromHeader(req)
+    if(token == ""){
+        sendResponse(res, "No Token Found", "INVALID TOKEN")
+        return
+    }
+    const url = `${uri}?access_token=${token}`
+    let headers = new Map()
+    headers.set("Authorization", `Bearer ${token}`)
+    const [status, response, err] = await externalApiCall('get', url, {}, headers)
+ 
+    if(err != null){
+        sendResponse(res, "", err)
+        return
+    }
+    if(status != 200){
+        sendResponse(res, "Token Expired", "INVALID TOKEN")
+        return
+    }
+    req.userInfo = response
+    next()
+}
+
+export function verifyUser(checkIsActive, checkKyc, checkIsWalletLiked, checkIsCustodialWalletLiked) {
+    return async function(req, res, next){
+        var [user, err] = await getUserByEmail(req.userInfo.email)
```

Comment:
verify User should also include verify Google Token flow. 
If in future, if verify User has all the validation checks , then it should also check whether the token passed is valid or not.